pennylane/operation.py

```diff
@@ -481,66 +481,74 @@ def matrix(self, wire_order=None):
         return expand_matrix(canonical_matrix, wires=self.wires, wire_order=wire_order)
 
     @classmethod
-    def _eigvals(cls, *params):
-        """Eigenvalues of the operator.
+    def compute_eigvals(cls, *params, **hyperparams):
+        """Canonical eigenvalues of the operator in the computational basis.
 
-        This is a *class method* that should be defined for all
-        new operations and observables that returns the eigenvalues
-        of the operator.
+        The canonical eigenvalues refer to the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        This class method allows eigenvalues to be computed
+        directly without instantiating the operator first.
+
+        The default implementation relies on the presence of the
+        :attr:`~.Operator.compute_matrix` method. To return the eigenvalues of *instantiated* operators,
+        please use the :attr:`~.Operator.eigvals()` method instead.
 
         If :attr:`diagonalizing_gates` are specified, the order of the
-        eigenvalues needs to match the order of
+        eigenvalues matches the order of
         the computational basis vectors when the observable is
         diagonalized using these ops. Otherwise, no particular order is
         guaranteed.
 
-        This private method allows eigenvalues to be computed
-        directly without instantiating the operators first.
+        .. note::
+            This method gets overwritten by subclasses to define the eigenvalues
+            of a particular operator.
 
-        The default implementation relies on the presence of the
-        :attr:`compute_matrix` method.
+        Args:
+            params (list): trainable parameters of this operator, as stored in ``op.parameters``
+            hyperparams (dict): non-trainable hyperparameters of this operator, as stored in ``op.hyperparameters``
 
-        To return the eigenvalues of *instantiated* operators,
-        please use the :attr:`~.Operator.eigvals` property instead.
+        Returns:
+            array: eigenvalues
 
         **Example:**
 
-        >>> qml.RZ._eigvals(0.5)
+        >>> qml.RZ.compute_eigvals(0.5)
         array([0.96891242-0.24740396j, 0.96891242+0.24740396j])
         >>> qml.PauliX(wires=0).diagonalizing_gates()
         [Hadamard(wires=[0])]
-        >>> qml.PauliX._eigvals()
+        >>> qml.PauliX.compute_eigvals()
         array([1, -1])
-
-        Returns:
-            array: eigenvalue representation
         """
-        return np.linalg.eigvals(cls.compute_matrix(*params))
+        return np.linalg.eigvals(cls.compute_matrix(*params, **hyperparams))
```

Comment:
can we replace this with `qml.math.linalg.eigvals`?

Reply:
should we?

Reply:
Maybe not right now? It sounds like a bigger unscheduled change?